src/SingleTableInheritanceTrait.php

```diff
@@ -63,6 +64,9 @@ public static function getSingleTableTypeMap() {
       // prevent infinite recursion if the singleTableSubclasses is inherited
       if (!in_array($calledClass, $subclasses)) {
         foreach ($subclasses as $subclass) {
+          if(!is_subclass_of($subclass, $calledClass)) {
+            throw new SingleTableInheritanceWrongInheritanceException('Subclass must extend its parent class. Parent: '.$calledClass.' | Subclass: '.$subclass);
+          }
```

Comment:
a) Should SingleTableInheritanceWrongInheritanceException take the two classes and generate the message itself?
b) let's use String interpolation vs concatenation

Reply:
a) Can you provide an example for me? I need your help.
b) Got it.

This is my best for now.
```php
          if(!is_subclass_of($subclass, $calledClass)) {
            throw new SingleTableInheritanceWrongInheritanceException("Subclass must extend its parent class. Parent: $calledClass | Subclass: $subclass");
          }
```